temporal-serviceclient/src/main/java/io/temporal/serviceclient/WorkflowServiceStubsImpl.java

```diff
@@ -138,7 +140,7 @@ public WorkflowServiceStubsImpl(
         new GrpcMetricsInterceptor(options.getMetricsScope());
     ClientInterceptor deadlineInterceptor = new GrpcDeadlineInterceptor(options);
     GrpcTracingInterceptor tracingInterceptor = new GrpcTracingInterceptor();
-    Metadata headers = new Metadata();
+    Metadata headers = headersFrom(options.getHeaders());
```

Comment:
It looks like this wouldn't allow binary headers. Should we just change options.headers type to io.grpc.Metadata?

Reply:
Good call, let me do that, it would give more flexibility on the client side. Also factory options are tied to grpc anyways so exposing some grpc objects externally wouldn't be too bad.